pkg/controller/case_test.go

```diff
@@ -934,6 +958,670 @@ func (ct *controllerTest) v1Now() *metav1.Time {
+	})
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf(
+			"instance with proper conditions not found, the existing conditions: %+v", lastInstance.Status.Conditions)
+	}
+	return err
+}
+
+// WaitForServiceInstanceProcessedGeneration waits until ServiceInstance parameter `Status.ObservedGeneration` is
+// equal or higher than ServiceInstance `generation` value, ServiceInstance is in Ready/True status and
+// ServiceInstance is not in Orphan Mitigation progress
+func (ct *controllerTest) WaitForServiceInstanceProcessedGeneration(generation int64) error {
+	err := wait.PollImmediate(pollingInterval, pollingTimeout, func() (bool, error) {
+		instance, err := ct.scInterface.ServiceInstances(testNamespace).Get(testServiceInstanceName, v1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting Instance: %v", err)
+		}
+
+		if instance.Status.ObservedGeneration >= generation &&
+			isServiceInstanceConditionTrue(instance) &&
+			!instance.Status.OrphanMitigationInProgress {
+			return true, nil
+		}
+
+		return false, nil
+	})
+
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf(
+			"instance with proper ProcessedGeneration status not found")
+	}
+	return err
+}
+
+func isServiceInstanceConditionTrue(instance *v1beta1.ServiceInstance) bool {
+	for _, cond := range instance.Status.Conditions {
+		if cond.Type == v1beta1.ServiceInstanceConditionReady || cond.Type == v1beta1.ServiceInstanceConditionFailed {
+			return cond.Status == v1beta1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+// AssertServiceInstanceHasNoCondition makes sure ServiceInstance is in not specific condition
+func (ct *controllerTest) AssertServiceInstanceHasNoCondition(t *testing.T, cond v1beta1.ServiceInstanceCondition) {
+	instance, err := ct.scInterface.ServiceInstances(testNamespace).Get(testServiceInstanceName, v1.GetOptions{})
+	if err != nil {
+		t.Fatalf("error getting Instance: %v", err)
+	}
+
+	for _, condition := range instance.Status.Conditions {
+		if t1, t2 := condition.Type, cond.Type; t1 == t2 {
+			if s1, s2 := condition.Status, cond.Status; s1 == s2 {
+				t.Fatalf(
+					"unexpected condition status: expected %v, got %v or \n "+
+						"unexpected condition type: expected %v, got %v", s2, s1, t2, t1)
+			}
+		}
+	}
+}
+
+// AssertServiceInstanceOrphanMitigationStatus makes sure ServiceInstance is/or is not in Orphan Mitigation progress
+func (ct *controllerTest) AssertServiceInstanceOrphanMitigationStatus(t *testing.T, state bool) {
+	instance, err := ct.scInterface.ServiceInstances(testNamespace).Get(testServiceInstanceName, v1.GetOptions{})
+	if err != nil {
+		t.Fatalf("error getting Instance: %v", err)
+	}
+
+	if om := instance.Status.OrphanMitigationInProgress; om != state {
+		t.Fatalf("unexpected OrphanMitigationInProgress status: expected %v, got %v", state, om)
+	}
+}
+
+// CreateClusterServiceClass creates ClusterServiceClass with default parameters
+func (ct *controllerTest) CreateClusterServiceClass() error {
```

Comment:
Pretty much all of this might want to be moved into test/util. I could have sworn there was an equivalent for this in there already but I can't find it offhand.